File: backend/src/main/java/magazyn/dto/DostawcaDTO.java
package magazyn.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DostawcaDTO {

    private Integer id;

    @NotBlank(message = "Nazwa dostawcy jest wymagana")
    private String nazwaDostawcy;

    @NotBlank(message = "Adres jest wymagany")
    private String adres;

    @NotBlank(message = "Numer telefonu jest wymagany")
    private String telefon;
}

File: frontend/app/src/main/java/com/example/magazyn/api/dtos/DostawcaDTO.kt
package com.example.magazyn.api.dtos

import com.google.gson.annotations.SerializedName

data class DostawcaDTO(
    val id: Int,
    @SerializedName("nazwaDostawcy")
    val nazwa: String,
    val adres: String,
    val telefon: String
)